Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1294,7 +1294,7 @@ class BrowserViewController: UIViewController {
         if let homePanelController = homePanelController {
             return .HomePanels(homePanelState: homePanelController.homePanelState)
         }
-        guard let tab = tabManager.selectedTab where !tab.loading else {
+        guard let tab = tabManager.selectedTab else {
             return .Loading
```

Comment:
What does the `.Loading` state mean? Is it useful at all?

I'm still not sure we've paid down the technical debt around `AppState` and our half in half out implementation of redux.

The `isPrivate` should probably live apart from the state of the tab, and `.Loading` should be a property of the `TabState`.

Most of the time, `.Loading` is irrelevant or an add-on to `.TabState`.

I think this should do, but could you raise a bug to rationalize the UI state into a more formal Redux architecture.